### DISCUSSION TITLE
importlib.metadata.PackageNotFoundError: No package metadata was found for meson

### DISCUSSION THREAD

**Discussion by tz0616** [Homebrew/discussions#5008] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ff899e5b3ff6731208564ef502db237e2627e8ba
Last commit: 5 days ago
Core tap JSON: 24 Dec 02:54 UTC
HOMEBREW_PREFIX: /usr/local/homebrew
HOMEBREW_REPOSITORY: /usr/local/homebrew
HOMEBREW_CELLAR: /usr/local/homebrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0.0
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-x86_64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.1
```

### Output of `brew doctor`

```shell
brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: gettext files detected at a system prefix.
These files can cause compilation and link failures, especially if they
are compiled with improper architectures. Consider removing these files:
  /usr/local/homebrew/lib/libgettextlib.dylib
  /usr/local/homebrew/lib/libintl.dylib
  /usr/local/homebrew/include/libintl.h

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libgcc_ext.10.4.dylib
  /usr/local/lib/libgcc_ext.10.5.dylib
  /usr/local/lib/libgcc_s.1.dylib
  /usr/local/lib/libgfortran.4.dylib
  /usr/local/lib/libgomp.1.dylib
  /usr/local/lib/libhdf5.8.dylib
  /usr/local/lib/libhdf5_hl.8.dylib
  /usr/local/lib/libnetcdf.18.dylib
  /usr/local/lib/libnetcdff.7.dylib
  /usr/local/lib/libquadmath.0.dylib
  /usr/local/lib/librtmp.dylib
  /usr/local/lib/libstdc++.6.dylib
  /usr/local/lib/libsz.2.0.0.dylib
  /usr/local/lib/libtcl8.6.dylib
  /usr/local/lib/libtk8.6.dylib
  /usr/local/lib/libz.1.2.8.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/H5ACpublic.h
  /usr/local/include/H5Apublic.h
  /usr/local/include/H5Cpublic.h
  /usr/local/include/H5DOpublic.h
  /usr/local/include/H5DSpublic.h
  /usr/local/include/H5Dpublic.h
  /usr/local/include/H5Epubgen.h
  /usr/local/include/H5Epublic.h
  /usr/local/include/H5FDcore.h
  /usr/local/include/H5FDdirect.h
  /usr/local/include/H5FDfamily.h
  /usr/local/include/H5FDlog.h
  /usr/local/include/H5FDmpi.h
  /usr/local/include/H5FDmpio.h
  /usr/local/include/H5FDmulti.h
  /usr/local/include/H5FDpublic.h
  /usr/local/include/H5FDsec2.h
  /usr/local/include/H5FDstdio.h
  /usr/local/include/H5Fpublic.h
  /usr/local/include/H5Gpublic.h
  /usr/local/include/H5IMpublic.h
  /usr/local/include/H5Ipublic.h
  /usr/local/include/H5LTpublic.h
  /usr/local/include/H5Lpublic.h
  /usr/local/include/H5MMpublic.h
  /usr/local/include/H5Opublic.h
  /usr/local/include/H5PLextern.h
  /usr/local/include/H5PTpublic.h
  /usr/local/include/H5Ppublic.h
  /usr/local/include/H5Rpublic.h
  /usr/local/include/H5Spublic.h
  /usr/local/include/H5TBpublic.h
  /usr/local/include/H5Tpublic.h
  /usr/local/include/H5Zpublic.h
  /usr/local/include/H5api_adpt.h
  /usr/local/include/H5overflow.h
  /usr/local/include/H5pubconf.h
  /usr/local/include/H5public.h
  /usr/local/include/H5version.h
  /usr/local/include/fakemysql.h
  /usr/local/include/fakepq.h
  /usr/local/include/fakesql.h
  /usr/local/include/hdf5.h
  /usr/local/include/hdf5_hl.h
  /usr/local/include/itcl.h
  /usr/local/include/itcl2TclOO.h
  /usr/local/include/itclDecls.h
  /usr/local/include/itclInt.h
  /usr/local/include/itclIntDecls.h
  /usr/local/include/itclMigrate2TclCore.h
  /usr/local/include/itclTclIntStubsFcn.h
  /usr/local/include/mysqlStubs.h
  /usr/local/include/netcdf.h
  /usr/local/include/netcdf_aux.h
  /usr/local/include/netcdf_dispatch.h
  /usr/local/include/netcdf_filter.h
  /usr/local/include/netcdf_mem.h
  /usr/local/include/netcdf_meta.h
  /usr/local/include/odbcStubs.h
  /usr/local/include/pqStubs.h
  /usr/local/include/ricehdf.h
  /usr/local/include/szip_adpt.h
  /usr/local/include/szlib.h
  /usr/local/include/tcl.h
  /usr/local/include/tclDecls.h
  /usr/local/include/tclOO.h
  /usr/local/include/tclOODecls.h
  /usr/local/include/tclPlatDecls.h
  /usr/local/include/tclThread.h
  /usr/local/include/tclTomMath.h
  /usr/local/include/tclTomMathDecls.h
  /usr/local/include/tdbc.h
  /usr/local/include/tdbcDecls.h
  /usr/local/include/tdbcInt.h
  /usr/local/include/tk.h
  /usr/local/include/tkDecls.h
  /usr/local/include/tkPlatDecls.h
  /usr/local/include/zconf.h
  /usr/local/include/zlib.h

Warning: Unbrewed '.la' files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.la' files:
  /usr/local/lib/libh5bzip2.la
  /usr/local/lib/libhdf5.la
  /usr/local/lib/libhdf5_hl.la
  /usr/local/lib/libnetcdf.la
  /usr/local/lib/libnetcdff.la
  /usr/local/lib/libsz.la

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/netcdf-fortran.pc
  /usr/local/lib/pkgconfig/netcdf.pc
  /usr/local/lib/pkgconfig/tcl.pc
  /usr/local/lib/pkgconfig/tk.pc
  /usr/local/lib/pkgconfig/zlib.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libhdf5.a
  /usr/local/lib/libhdf5_hl.a
  /usr/local/lib/libnetcdf.a
  /usr/local/lib/libnetcdff.a
  /usr/local/lib/libsz.a
  /usr/local/lib/libtclstub8.6.a
  /usr/local/lib/libtkstub8.6.a
  /usr/local/lib/libz.a

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  meson

Warning: Your Homebrew's prefix is not /usr/local.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

Hi,
 The installation fails with the following errors.
 Do you know how to fix it?

 Thanks
-----------------------------
brew install gnutls openconnect openssl 
==> Downloading https://formulae.brew.sh/api/formula.jws.json
################################################################################################################################ 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
################################################################################################################################ 100.0%
Warning: openssl@3 3.2.0_1 is already installed and up-to-date.
To reinstall 3.2.0_1, run:
  brew reinstall openssl@3
==> Fetching dependencies for gnutls: p11-kit, libevent, libnghttp2 and unbound
==> Fetching p11-kit
Warning: Building p11-kit from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /usr/local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /usr/local/homebrew)
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/a356f125c366e6c01ff5a0bfb0f1a83f954b9579/Formula/p/p11-kit.rb
################################################################################################################################ 100.0%
==> Downloading https://github.com/p11-glue/p11-kit/releases/download/0.25.3/p11-kit-0.25.3.tar.xz
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/5fda7cc5095f391f971d4f418d5ec2e82a39829ffbb349b14e9a74e58cdb4f68--p11-kit-0.25.3.tar.xz
==> Fetching libevent
==> Downloading https://ghcr.io/v2/homebrew/core/libevent/manifests/2.1.12_1
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/68b113f9ab63db45f4e1860de522ce2ca4fa081eb3c0d5c7d6005a35c3cf8d06--libevent-2.1.12_1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/libevent/blobs/sha256:5d54f13cd93d87185bd7bb592cb945d1f64cac3e88d1e46c2fb5d9ea538d9623
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/04f847fd8a3e396a3957fc7c95e77d47c292140c4dc8e2795dd8dedf9a307478--libevent--2.1.12_1.sonoma.bottle.tar.gz
==> Fetching libnghttp2
==> Downloading https://ghcr.io/v2/homebrew/core/libnghttp2/manifests/1.58.0
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/9b2081d73959d4308c84740344d5f8169bbb9125dd76d20a71556a11a0171253--libnghttp2-1.58.0.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/libnghttp2/blobs/sha256:14a08800cac734d4b2722864ba2b03ae786aa5dfee3d224ee71b2c8bd88b16
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/cbc3eee18fd92f1a54656dccbb053d7d546fb90132e062ad5886cdbdc4239c19--libnghttp2--1.58.0.sonoma.bottle.tar.gz
==> Fetching unbound
Warning: Building unbound from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /usr/local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /usr/local/homebrew)
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/a356f125c366e6c01ff5a0bfb0f1a83f954b9579/Formula/u/unbound.rb
################################################################################################################################ 100.0%
==> Downloading https://nlnetlabs.nl/downloads/unbound/unbound-1.19.0.tar.gz
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/254e568fdf37020292b85d36ba425f16eb7d811cf42b0aa2b3b6862cd0fb64d1--unbound-1.19.0.tar.gz
==> Fetching gnutls
Warning: Building gnutls from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /usr/local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /usr/local/homebrew)
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/a356f125c366e6c01ff5a0bfb0f1a83f954b9579/Formula/g/gnutls.rb
################################################################################################################################ 100.0%
==> Downloading https://www.gnupg.org/ftp/gcrypt/gnutls/v3.8/gnutls-3.8.2.tar.xz
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/71b8292530a732bba1ea94a35f8c3977c5cb1b780e81b6e1786152a2eb372033--gnutls-3.8.2.tar.xz
==> Fetching dependencies for openconnect: p11-kit, unbound, gnutls and stoken
==> Fetching stoken
==> Downloading https://ghcr.io/v2/homebrew/core/stoken/manifests/0.93
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/7ef6178c25578d41919bd9775a1ea0c77333661c7194aa0f574525ace45a278d--stoken-0.93.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/stoken/blobs/sha256:8b6a3bb0bc8d471162e70570a9effdc056f007d43eeb6974b452518fe6c4162f
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/d7c4ad1161b42bd87553e5085d353a2aaf13622c321ddceeae7198d55f22a86f--stoken--0.93.sonoma.bottle.tar.gz
==> Fetching openconnect
Warning: Building openconnect from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /usr/local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /usr/local/homebrew)
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/a356f125c366e6c01ff5a0bfb0f1a83f954b9579/Formula/o/openconnect
################################################################################################################################ 100.0%
==> Downloading https://gitlab.com/openconnect/openconnect/-/commit/7512698217c4104aade7a2df669a20de68f3bb8c.diff
##O#-   #                                                                                                                             
==> Downloading https://gitlab.com/openconnect/vpnc-scripts/-/raw/473d3e810b8fe8223058ab580fac20c53204e677/vpnc-script
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/1b7f88a9790d37a4934e0a751544ee70c40e345c1b32d97c23da07c45f3a8162--vpnc-script
==> Downloading https://www.infradead.org/openconnect/download/openconnect-9.12.tar.gz
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/cb27d1442fcaee671e3cbf37db09ef51faac0653757a21fa9df88854fc7f3471--openconnect-9.12.tar.gz
==> Installing dependencies for gnutls: p11-kit, libevent, libnghttp2 and unbound
==> Installing gnutls dependency: p11-kit
==> meson setup build -Dsystem_config=/usr/local/homebrew/etc -Dmodule_config=/usr/local/homebrew/etc/pkcs11/modules -Dtrust_paths=/usr
Last 15 lines from /Users/tzhang/Library/Logs/Homebrew/p11-kit/01.meson:
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/homebrew/opt/meson/bin/meson", line 33, in <module>
    sys.exit(load_entry_point('meson==1.3.0', 'console_scripts', 'meson')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/homebrew/opt/meson/bin/meson", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Homebrew/Cellar/python@3.12/3.12.1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/metadata/__init__.py", line 861, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Homebrew/Cellar/python@3.12/3.12.1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/metadata/__init__.py", line 399, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for meson

Do not report this issue to Homebrew/brew or Homebrew/homebrew-core!

**Comment by branchv** [comment]
```
Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  meson
```

That's the problem, you may have a pip-installed `meson` that caused the link failure. You can fix with `brew link --overwrite meson`

**Reply by tz0616** [reply]
Hi branchvincent,
Thanks for your comments. 
 There are still new errors. see below
 Do you know why?
 Thanks
---------------------
brew link --overwrite meson
Linking /usr/local/homebrew/Cellar/meson/1.3.0... 243 symlinks created.

-----------------
 brew install gnutls openconnect openssl 
==> Downloading https://formulae.brew.sh/api/formula.jws.json
################################################################################################################################# 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
################################################################################################################################# 100.0%
Warning: openssl@3 3.2.0_1 is already installed and up-to-date.
To reinstall 3.2.0_1, run:
  brew reinstall openssl@3
==> Fetching dependencies for gnutls: p11-kit, libevent, libnghttp2 and unbound
==> Fetching p11-kit
Warning: Building p11-kit from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /usr/local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /usr/local/homebrew)
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/a356f125c366e6c01ff5a0bfb0f1a83f954b9579/Formula/p/p11-kit.rb
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/a4f7944bbf48153052a0ca84604a06f523629423d63f6ce11f1798de24c6f168--p11-kit.rb
==> Downloading https://github.com/p11-glue/p11-kit/releases/download/0.25.3/p11-kit-0.25.3.tar.xz
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/5fda7cc5095f391f971d4f418d5ec2e82a39829ffbb349b14e9a74e58cdb4f68--p11-kit-0.25.3.tar.xz
==> Fetching libevent
==> Downloading https://ghcr.io/v2/homebrew/core/libevent/manifests/2.1.12_1
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/68b113f9ab63db45f4e1860de522ce2ca4fa081eb3c0d5c7d6005a35c3cf8d06--libevent-2.1.12_1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/libevent/blobs/sha256:5d54f13cd93d87185bd7bb592cb945d1f64cac3e88d1e46c2fb5d9ea538d9623
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/04f847fd8a3e396a3957fc7c95e77d47c292140c4dc8e2795dd8dedf9a307478--libevent--2.1.12_1.sonoma.bottle.tar.gz
==> Fetching libnghttp2
==> Downloading https://ghcr.io/v2/homebrew/core/libnghttp2/manifests/1.58.0
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/9b2081d73959d4308c84740344d5f8169bbb9125dd76d20a71556a11a0171253--libnghttp2-1.58.0.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/libnghttp2/blobs/sha256:14a08800cac734d4b2722864ba2b03ae786aa5dfee3d224ee71b2c8bd88b16c
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/cbc3eee18fd92f1a54656dccbb053d7d546fb90132e062ad5886cdbdc4239c19--libnghttp2--1.58.0.sonoma.bottle.tar.gz
==> Fetching unbound
Warning: Building unbound from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /usr/local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /usr/local/homebrew)
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/a356f125c366e6c01ff5a0bfb0f1a83f954b9579/Formula/u/unbound.rb
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/702f337bb2be75aa99de3a625706a2514819e774e7ec3ceaa8a1a903066b380f--unbound.rb
==> Downloading https://nlnetlabs.nl/downloads/unbound/unbound-1.19.0.tar.gz
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/254e568fdf37020292b85d36ba425f16eb7d811cf42b0aa2b3b6862cd0fb64d1--unbound-1.19.0.tar.gz
==> Fetching gnutls
Warning: Building gnutls from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /usr/local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /usr/local/homebrew)
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/a356f125c366e6c01ff5a0bfb0f1a83f954b9579/Formula/g/gnutls.rb
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/3d88ef678e7bcb27a77c23e79764442a291c6a78b9dd615f71d5fd94152b4232--gnutls.rb
==> Downloading https://www.gnupg.org/ftp/gcrypt/gnutls/v3.8/gnutls-3.8.2.tar.xz
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/71b8292530a732bba1ea94a35f8c3977c5cb1b780e81b6e1786152a2eb372033--gnutls-3.8.2.tar.xz
==> Fetching dependencies for openconnect: p11-kit, unbound, gnutls and stoken
==> Fetching stoken
==> Downloading https://ghcr.io/v2/homebrew/core/stoken/manifests/0.93
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/7ef6178c25578d41919bd9775a1ea0c77333661c7194aa0f574525ace45a278d--stoken-0.93.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/stoken/blobs/sha256:8b6a3bb0bc8d471162e70570a9effdc056f007d43eeb6974b452518fe6c4162f
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/d7c4ad1161b42bd87553e5085d353a2aaf13622c321ddceeae7198d55f22a86f--stoken--0.93.sonoma.bottle.tar.gz
==> Fetching openconnect
Warning: Building openconnect from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /usr/local/homebrew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /usr/local/homebrew)
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/a356f125c366e6c01ff5a0bfb0f1a83f954b9579/Formula/o/openconnect.
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/3f566dc6ab8ee26dbcb8edfd10526a81f067b457df497b3459b8bb348bd4134c--openconnect.rb
==> Downloading https://gitlab.com/openconnect/openconnect/-/commit/7512698217c4104aade7a2df669a20de68f3bb8c.diff
##O#-   #                                                                                                                              
==> Downloading https://gitlab.com/openconnect/vpnc-scripts/-/raw/473d3e810b8fe8223058ab580fac20c53204e677/vpnc-script
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/1b7f88a9790d37a4934e0a751544ee70c40e345c1b32d97c23da07c45f3a8162--vpnc-script
==> Downloading https://www.infradead.org/openconnect/download/openconnect-9.12.tar.gz
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/cb27d1442fcaee671e3cbf37db09ef51faac0653757a21fa9df88854fc7f3471--openconnect-9.12.tar.gz
==> Installing dependencies for gnutls: p11-kit, libevent, libnghttp2 and unbound
==> Installing gnutls dependency: p11-kit
==> meson setup build -Dsystem_config=/usr/local/homebrew/etc -Dmodule_config=/usr/local/homebrew/etc/pkcs11/modules -Dtrust_paths=/usr/
==> meson compile -C build --verbose
==> meson test -C build
==> meson install -C build
🍺  /usr/local/homebrew/Cellar/p11-kit/0.25.3: 26 files, 3.8MB, built in 43 seconds
==> Installing gnutls dependency: libevent
==> Downloading https://ghcr.io/v2/homebrew/core/libevent/manifests/2.1.12_1
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/68b113f9ab63db45f4e1860de522ce2ca4fa081eb3c0d5c7d6005a35c3cf8d06--libevent-2.1.12_1.bottle_manifest.json
==> Pouring libevent--2.1.12_1.sonoma.bottle.tar.gz
🍺  /usr/local/homebrew/Cellar/libevent/2.1.12_1: 57 files, 2MB
==> Installing gnutls dependency: libnghttp2
==> Downloading https://ghcr.io/v2/homebrew/core/libnghttp2/manifests/1.58.0
Already downloaded: /Users/tzhang/Library/Caches/Homebrew/downloads/9b2081d73959d4308c84740344d5f8169bbb9125dd76d20a71556a11a0171253--libnghttp2-1.58.0.bottle_manifest.json
==> Pouring libnghttp2--1.58.0.sonoma.bottle.tar.gz
🍺  /usr/local/homebrew/Cellar/libnghttp2/1.58.0: 13 files, 722KB
==> Installing gnutls dependency: unbound
==> ./configure --sysconfdir=/usr/local/homebrew/etc --enable-event-api --enable-tfo-client --enable-tfo-server --with-libevent=/usr/loc
==> make
==> make install
🍺  /usr/local/homebrew/Cellar/unbound/1.19.0: 57 files, 5.6MB, built in 1 minute 19 seconds
==> Installing gnutls
==> ./configure --disable-silent-rules --disable-static --sysconfdir=/usr/local/homebrew/etc --with-default-trust-store-file=/usr/local/
==> make install
Last 15 lines from /Users/tzhang/Library/Logs/Homebrew/gnutls/02.make:
1 error generated.
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I../.. -I./../../gl -I./../../gl -I./../includes -I./../includes -I./.. -Wtype-limits -fno-common -Wall -Wbad-function-cast -Wdate-time -Wdisabled-optimization -Wdouble-promotion -Wextra -Winit-self -Winvalid-pch -Wmissing-declarations -Wmissing-include-dirs -Wmissing-prototypes -Wnested-externs -Wnull-dereference -Wold-style-definition -Wpacked -Wpointer-arith -Wshadow -Wstrict-prototypes -Wuninitialized -Wunknown-pragmas -Wvariadic-macros -Wwrite-strings -Wformat=2 -Wno-missing-field-initializers -Wno-unused-parameter -fdiagnostics-show-option -fno-builtin-strcmp -I/usr/local/homebrew/Cellar/nettle/3.9.1/include -I/usr/local/homebrew/Cellar/libtasn1/4.19.0/include -I/usr/local/homebrew/Cellar/libidn2/2.3.4_1/include -I/usr/local/homebrew/Cellar/p11-kit/0.25.3/include/p11-kit-1 -g -O2 -c email-verify.c  -fno-common -DPIC -o .libs/email-verify.o
make[4]: *** [x509.lo] Error 1
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I../.. -I./../../gl -I./../../gl -I./../includes -I./../includes -I./.. -Wtype-limits -fno-common -Wall -Wbad-function-cast -Wdate-time -Wdisabled-optimization -Wdouble-promotion -Wextra -Winit-self -Winvalid-pch -Wmissing-declarations -Wmissing-include-dirs -Wmissing-prototypes -Wnested-externs -Wnull-dereference -Wold-style-definition -Wpacked -Wpointer-arith -Wshadow -Wstrict-prototypes -Wuninitialized -Wunknown-pragmas -Wvariadic-macros -Wwrite-strings -Wformat=2 -Wno-missing-field-initializers -Wno-unused-parameter -fdiagnostics-show-option -fno-builtin-strcmp -I/usr/local/homebrew/Cellar/nettle/3.9.1/include -I/usr/local/homebrew/Cellar/libtasn1/4.19.0/include -I/usr/local/homebrew/Cellar/libidn2/2.3.4_1/include -I/usr/local/homebrew/Cellar/p11-kit/0.25.3/include/p11-kit-1 -g -O2 -c pkcs7-output.c  -fno-common -DPIC -o .libs/pkcs7-output.o
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I../.. -I./../../gl -I./../../gl -I./../includes -I./../includes -I./.. -Wtype-limits -fno-common -Wall -Wbad-function-cast -Wdate-time -Wdisabled-optimization -Wdouble-promotion -Wextra -Winit-self -Winvalid-pch -Wmissing-declarations -Wmissing-include-dirs -Wmissing-prototypes -Wnested-externs -Wnull-dereference -Wold-style-definition -Wpacked -Wpointer-arith -Wshadow -Wstrict-prototypes -Wuninitialized -Wunknown-pragmas -Wvariadic-macros -Wwrite-strings -Wformat=2 -Wno-missing-field-initializers -Wno-unused-parameter -fdiagnostics-show-option -fno-builtin-strcmp -I/usr/local/homebrew/Cellar/nettle/3.9.1/include -I/usr/local/homebrew/Cellar/libtasn1/4.19.0/include -I/usr/local/homebrew/Cellar/libidn2/2.3.4_1/include -I/usr/local/homebrew/Cellar/p11-kit/0.25.3/include/p11-kit-1 -g -O2 -c virt-san.c  -fno-common -DPIC -o .libs/virt-san.o
In file included from verify-high2.c:37:
./../pkcs11_int.h:27:10: fatal error: 'p11-kit/pkcs11.h' file not found
#include <p11-kit/pkcs11.h>
         ^~~~~~~~~~~~~~~~~~
1 error generated.
make[4]: *** [verify-high2.lo] Error 1
make[3]: *** [install] Error 2
make[2]: *** [install-recursive] Error 1
make[1]: *** [install] Error 2
make: *** [install-recursive] Error 1

Do not report this issue to Homebrew/brew or Homebrew/homebrew-core!

**Reply by branchv** [reply]
Not sure about the exact build failure but:

```
Warning: Building openconnect from source as the bottle needs:
HOMEBREW_PREFIX: /usr/local (yours is /usr/local/homebrew)
```

This is strange, did you mean to install homebrew in a non-standard prefix? It's not supported and will likely cause build issues like this.

**Reply by tz0616** [reply]
I download the .pkg from the website. I do not see the option to select the installation path.

**Reply by tz0616** [reply]
I get .pkg from https://github.com/Homebrew/brew/releases/tag/4.2.0
The website  (https://brew.sh/)   said "Homebrew won’t install files outside its prefix and you can place a Homebrew installation wherever you like."

**Reply by branchv** [reply]
Ah I assume that's a bug in the new .pkg then, can you open an issue at https://github.com/Homebrew/brew/issues/new/choose? 

For now, I'd recommend re-installing with the official installer: https://github.com/Homebrew/install

> you can place a Homebrew installation wherever you like

This is true but https://docs.brew.sh/FAQ#why-should-i-install-homebrew-in-the-default-location explains why it's a better experience to use the default prefix

**Reply by tz0616** [reply]
I can not install the command line, see below. I use "csh", I don't know why it did not work
/usr/local> /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
Illegal variable name.

**Reply by branchv** [reply]
Open a bash shell first and then try again

```
$ bash
$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

**Reply by tz0616** [reply]
Could you check What's your installation location of homebrew? I am not sure how to uninstall the present homebrew and all packages in my  Mac because I installed homebrew via ".pkg" file.
